### ISSUE TITLE
Refactor `SyncingCoordinator` to fix infinite scroll where the response size does not match the list view size

### ISSUE THREAD

**Issue by jaclync** [woocommerce/woocommerce-ios#2738]
Ref: p91TBi-30X-p2 - plan: p91TBi-30X#comment-1291-p2

The main work is to separate `SyncingCoordinator` into two components: scroll watcher and pagination tracker.

The first step is to fix an edge case when editing a grouped product's linked products with steps to reproduce:

Prerequisite: the store has a grouped product with more than 25 linked products
- Go to the Products tab
- Tap on the grouped product that has more than 25 linked products
- Tap on the "Grouped products" row
- Tap "X" to delete a linked product from the list
- Scroll down to the end of the list --> the infinite scroll does not work (if you didn't delete a linked product, the infinite scroll works)

After the issue above is resolved with the new infinite scroll architecture, we can start replacing other `SyncingCoordinator` usage in the code base.

**Comment by jaclync** [comment]
Now that the grouped products use case is fixed in https://github.com/woocommerce/woocommerce-ios/pull/2750 with `PaginationTracker` and `PaginationTrackerDelegate`, let's replace the rest in the future after Products M3.

**Comment by joshheald** [comment]
Automated triage note: this was tested and commented by Codex using device/simulator automation. I checked the current linked-products path in the WooCommerce iOS worktree using an iPhone 15 Pro simulator running iOS 17.5 as the target context. The original grouped-products linked-product selector now goes through PaginatedListSelectorViewController and PaginationTrackerDelegate rather than SyncingCoordinator. There are still other SyncingCoordinator usages in the app, but the specific infinite-scroll repro path from this issue appears to have been superseded by the pagination tracker work, so I would treat this issue as no longer relevant for the original bug.

I am leaving this GitHub issue open and applying the `triage: automation attempted` label for follow-up visibility.

**Comment by dangermattic** [comment]
🚫 Please add a type label (e.g. **type: enhancement**) and a feature label (e.g. **feature: stats**) to this issue.

**Comment by jaclync** [comment]
Closing, the main issue was fixed.
